### PR TITLE
Adding a getFile function to fetch content of a file

### PR DESCRIPTION
**Pull request by shruthi-venkateswaran**

### CODE REVIEW

**Comment by nkatzman** [comment]
Dependent on: https://github.com/screwdriver-cd/data-schema/pull/28

**Comment by nkatzman** [comment]
👍

File: test/index.test.js [hunk 1]
```diff
@@ -157,4 +162,33 @@ describe('index test', () => {
...
+        it('returns not implemented', () => {
+            const config = {
+                scmUrl: 'foo',
+                path: 'testFile',
+                token: 'token'
+            };
+
+            instance.getFile(config)
+              .then(() => {
+                  assert(false, 'you will never get dis');
```
Comment (petey) [inline test/index.test.js]: assert.fail()

File: index.js [hunk 2]
```diff
@@ -111,6 +111,33 @@ class ScmBase {
...
+    * @param  {String}   config.path         The file in the repo to fetch
+    * @param  {String}   config.token        The token used to authenticate to the SCM
+    * @return {Promise}
+    */
+    getFile(config) {
+        const result = Joi.validate(config, dataSchema.plugins.scm.getFile);
+
+        if (result.error) {
+            return new Promise((resolve, reject) => {
+                process.nextTick(() => reject(result.error));
```
Comment (d2lam) [inline index.js]: why do we need `process.nextTick`?

File: test/index.test.js [hunk 3]
```diff
@@ -157,4 +162,33 @@ describe('index test', () => {
...
+              })
+              .catch(err => {
+                  assert.isOk(err, 'Error should be returned');
```
Comment (petey) [inline test/index.test.js]: Maybe assert.instanceOf(err, Error); I find "isOk" to be a poor definition of an error.

File: index.js (hunk 2 above)
Comment (nkatzman) [inline index.js]: I'm not sure. Is it okay to reject the promise immediately?

Comment (petey) [inline index.js]: We've got code that resolves a promise immediately. :shrug:

Comment (nkatzman) [inline index.js]: A quick search online doesn't have anything bad. I read that promises are forced to run asynchronously

File: index.js [hunk 4]
```diff
@@ -44,18 +44,14 @@ class ScmBase {
         const result = Joi.validate(config, dataSchema.plugins.scm.getPermissions);
 
         if (result.error) {
-            return new Promise((resolve, reject) => {
-                process.nextTick(() => reject(result.error));
-            });
+            return Promise.reject(result.error);
```
Comment (d2lam) [inline index.js]: 👍

**Comment by d2lam** [comment]
Looks fine  👍

**Comment by petey** [comment]
👍

**Pull request merged**